platform/solr/solr-dependencies/pom.xml

```diff
@@ -21,199 +21,627 @@
+                            org.apache.solr.common,
+                            org.apache.solr.common.cloud,
+                            org.apache.solr.common.params,
+                            org.apache.solr.common.util,
+                            org.apache.solr.core,
+                            org.apache.solr.core.backup.repository,
+                            org.apache.solr.core.snapshots,
+                            org.apache.solr.handler,
+                            org.apache.solr.handler.admin,
+                            org.apache.solr.handler.component,
+                            org.apache.solr.logging,
+                            org.apache.solr.metrics,
+                            org.apache.solr.metrics.rrd,
+                            org.apache.solr.request,
+                            org.apache.solr.response,
+                            org.apache.solr.response.transform,
+                            org.apache.solr.rest,
+                            org.apache.solr.schema,
+                            org.apache.solr.search,
+                            org.apache.solr.search.stats,
+                            org.apache.solr.security,
+                            org.apache.solr.servlet,
+                            org.apache.solr.uninverting,
+                            org.apache.solr.update,
+                            org.apache.solr.update.processor,
+                            org.apache.solr.util,
+                            org.apache.solr.util.plugin,
+                            org.apache.zookeeper,
+                            org.apache.zookeeper.client,
+                            org.apache.zookeeper.data,
+                            org.apache.zookeeper.proto,
+                            org.noggit,
+                            org.rrd4j.core,
+
+                            <!-- used by solr-factory implementation -->
+                            org.apache.solr.client.solrj.embedded,
+
+                            <!-- used by  lux -->
+                            org.apache.lucene.analysis.charfilter,
+                            org.apache.lucene.analysis.core,
+                            org.apache.lucene.analysis.miscellaneous,
+                            org.apache.lucene.analysis.standard,
+                            org.apache.lucene.analysis.tokenattributes,
+                            org.apache.lucene.queryparser.classic,
+                            org.apache.lucene.queryparser.ext,
+                            org.apache.lucene.queryparser.xml,
+                            org.apache.lucene.search.highlight,
+                            org.apache.lucene.search.spans,
+                            org.apache.solr.parser,
+
+                            <!-- dependencies used by catalog-core-solr-->
+                            org.locationtech.*
+                        </Export-Package>
+                        <Embed-Dependency>
```

Comment:
It may be worth exporting these in the API that uses them and do Conditional-Package instead to see we don't really need all of these embedded dependencies.

Reply:
Looked into with @ryeats , went down quite the rabbit hole with it. It doesn't seem like it will be a cleaner solution.